### ISSUE TITLE
The redirect from www.example.com -> example.com does not work properly

### ISSUE THREAD

**Issue by ScuttleSE** [nickjj/ansible-nginx#15]
The server-segment in the config where the redirect from www.example.com to example.com is done does not contain any SSL-settings, leading nginx to totally freak out if you have multiple domains on the server. 

You need to add the same settings and same certs under both HTTPS server-sections in the conf

**Comment by nickjj** [comment]
It should work. It's what I'm running in production right now.

I have 3 domains on 1 server, all with their own vhost config, each redirecting from www to non-www.

**Comment by ScuttleSE** [comment]
Are they all using the same cert-file? In that case it would work fine, but if you have separate certs for the different domains nginx it won't

**Comment by nickjj** [comment]
Yes they are all using the same certificate, but each site also has its own vhost config, which in turn has its own SSL configuration, so I'm not seeing why that would cause issues for you.

**Comment by ScuttleSE** [comment]
Since the SSL handshake is done before any headers are loaded, Nginx will use the default SSL-cert on the site if nothing else is defined in the server-section. If that cert, like in your case, happen to have the current hostname everything is fine and dandy, but if the default ssl-cert is for foo.com and I load www.bar.com, things won't work.

**Comment by nickjj** [comment]
Shouldn't bar.com's vhost config have its own config which has bar's certificate? That redirect would also only be valid for bar.com, not foo.com (it wouldn't even match for foo.com).

**Comment by ScuttleSE** [comment]
Indeed it has, but if you look at the config-file, all of the SSL cert-config is enclosed in its on server-section for example.com. www.example.com doesn't know anything about that config, it will revert to whatever the default SSL certificate happen to be.

**Comment by ScuttleSE** [comment]
The easiest (only?) solution is to duplicate all the SSL-settings you have for bar.com in the server-section for www.bar.com

**Comment by nickjj** [comment]
Ok I see.

I forgot, the redirect is listening on port 443 (I thought it was on port 80), and the port 80 redirect to 443 is already happening prior to that in the Let's Encrypt challenge.

So you're saying it defaults to what you have set as the `default_server`?

**Comment by ScuttleSE** [comment]
Exactly. Or, in my case where I don't have a default_server, it seems to default to whichever cert it loads first when restarting...

In your case it doesn't make any difference since all the vhosts uses the same cert :)

**Comment by nickjj** [comment]
Since it's "just" a redirect, I wonder if we can get by with just duplicating the 3 SSL certificate values, rather than everything.

**Comment by ScuttleSE** [comment]
Probably, yeah, I'm gonna do some more testing once I can generate certs at letsencrypt again, accidently hit the rate limiter while testing here earlier :)

**Comment by nickjj** [comment]
For correctness and less "wtf is happening" it might be best to duplicate them all.

But please let me know if it works (if you duplicate them all) once your rate limit expires. That would be all of the `ssl_` values along with the resolver values. Probably the STS header too.

**Comment by ScuttleSE** [comment]
Yup, yup, I have already massacred your template-file to get the ssl working and to be able accommodate my "upstream only" servers :)

**Comment by nickjj** [comment]
In the end, having support for your modifications seems reasonable. It's not what I do personally, but I imagine a lot of other people would be interested.

And there's been times where I wanted to test something without potentially blowing up my SSL certificates on all domains, so maybe I'd start using that too.

If you get everything, can you email me your working configs to nick.janetakis@gmail.com? Then I'll implement, test and merge it into this role -- unless you're feeling generous and would like to do that via a PR.

**Comment by ScuttleSE** [comment]
A PR is definately doable, I'll get one up as soon as I get stuff running here

**Comment by nickjj** [comment]
Awesome thanks.
